pipelines/build/common/openjdk_build_pipeline.groovy

```diff
@@ -441,50 +441,45 @@ class Build {
-                                                filter: "**/${jobName}*.tap",
-                                                target: 'workspace/target/AQAvitTaps/',
-                                                fingerprintArtifacts: true,
-                                                flatten: true
-                                            )
-                                        }
-                                    } catch (Exception e) {
-                                        context.echo "Cannot run copyArtifacts from job ${jobName}. Exception: ${e.message}. Skipping copyArtifacts..."
-                                    }
-                                    context.archiveArtifacts artifacts: 'workspace/target/AQAvitTaps/*.tap', fingerprint: true
-                                } else {
-                                    context.echo "Warning: ${jobName} result is ${result}, no tap file is archived"
+
+                        def testJob = context.build job: jobName,
+                                        propagate: false,
+                                        parameters: [
+                                            context.string(name: 'UPSTREAM_JOB_NUMBER', value: "${env.BUILD_NUMBER}"),
+                                            context.string(name: 'UPSTREAM_JOB_NAME', value: "${env.JOB_NAME}"),
+                                            context.string(name: 'SDK_RESOURCE', value: 'upstream'),
+                                            context.string(name: 'JDK_REPO', value: jdkRepo),
+                                            context.string(name: 'JDK_BRANCH', value: jdkBranch),
+                                            context.string(name: 'OPENJ9_BRANCH', value: openj9Branch),
+                                            context.string(name: 'LABEL_ADDITION', value: additionalTestLabel),
+                                            context.booleanParam(name: 'KEEP_REPORTDIR', value: keep_test_reportdir),
+                                            context.string(name: 'PARALLEL', value: parallel),
+                                            context.string(name: 'NUM_MACHINES', value: "${numMachinesPerTest}"),
+                                            context.booleanParam(name: 'USE_TESTENV_PROPERTIES', value: useTestEnvProperties),
+                                            context.booleanParam(name: 'GENERATE_JOBS', value: aqaAutoGen),
+                                            context.string(name: 'ADOPTOPENJDK_BRANCH', value: aqaBranch),
+                                            context.string(name: 'ACTIVE_NODE_TIMEOUT', value: "${buildConfig.ACTIVE_NODE_TIMEOUT}"),
+                                            context.booleanParam(name: 'DYNAMIC_COMPILE', value: DYNAMIC_COMPILE)],
+                                        wait: true
+                        currentBuild.result = testJob.getResult()
```

Comment:
This is not going to propagate correctly as it will report the last testJob result for the build job, hence a Success would overwrite a Failure, also we ought to report a failure on copy artifact failure?

I think we need something like (copied from similar code [here](https://github.com/adoptium/ci-jenkins-pipelines/blob/4d6f5e2f22e7c5330da9791ef930f3ddf65624e8/pipelines/build/common/build_base_file.groovy#L888)):
```
                            if ((testJob.getResult() != 'SUCCESS' || !copyArtifactSuccess) && propagateFailures) {
                                context.error("Propagating downstream job result: ${jobName}, Result: "+testJob.getResult()+" CopyArtifactsSuccess: "+copyArtifactSuccess)
                                if (copyArtifactSuccess && testJob.getResult() == 'UNSTABLE' && (currentBuild.result == 'SUCCESS' || currentBuild.result == 'UNSTABLE' )) {
                                    currentBuild.result = 'UNSTABLE'
                                } else {
                                    currentBuild.result = 'FAILURE'
                                }
                            }

```

Reply:
Also, is this needed for this release?

Reply:
The job status can only be set to worse than it https://github.com/jenkinsci/jenkins/blob/213363d387736874f1d14d83e57347f757f3ed4f/core/src/main/java/hudson/model/Run.java#L462-L466. That is **Success would overwrite a Failure**.  I also verified here https://ci.adoptium.net/view/work-in-progress/job/Sophia_pipeline/327/console

Reply:
'copy artifact failure ' could happen if there is no artifacts to copy ( aborted or other exceptions), which is expected and has been reported by catch block
 ```catch (Exception e) {
                                context.echo "Cannot run copyArtifacts from job ${jobName}. Exception: ${e.message}. Skipping copyArtifacts..."
                            }```

Reply:
For aborted test jobs there might have artifacts ( parallel, one child job aborted, others good) or might not, which is reflected by test jobs status, so no need to report it specifically.

Reply:
Yes, I think both https://github.com/adoptium/ci-jenkins-pipelines/pull/681 and this PR are needed for release.

Reply:
Interesting, never knew that was a hidden behaviour of assigning to currentBuild.result, that is illogical! but as you say does seem to be what happens. To me if I code a statement to assign result to be Success, it should be! If I were to code otherwise i'd code as result = WorseOf(current, Success) not hide the magic!!
Anyway, so yes lets go with that.

Reply:
Yes, I was going to do same thing  like WorseThan() and found it's unnecessary, which I think it is reasonable as it's the same logic when using propagate=true. With propagate=true the build status will only be set if downstream job status is worse.